doc/lspconfig.txt

```diff
@@ -1,242 +1,659 @@
-9. Built-in commands                             *lspconfig-built-in-commands*
+INDEX                                                          *lspconfig-index*
+
+The autogenerated documentation for all language servers is accessible in
+`servers.md`. Type `gf` on the filename from this document to jump to the markdown
+file.
 
+==============================================================================
+ADDING NEW SERVERS                                    *lspconfig-adding-servers*
+
+`lspconfig` can also be used to add servers that are not available in the
+default configurations. This can be useful if there is an outstanding PR that
+is in review, or when developing a language server that is unavailable
+publicly.
+
+The three steps for adding a custom server are:
+
+- load the lspconfig module (note, this is a stylistic choice)
+>
+   local lspconfig = require 'lspconfig'
+<
+- define the configuration
+
+>
+   local configs = require 'lspconfig/configs'
+
+   -- Check if the config is already defined (useful when reloading this file)
+   if not lspconfig.foo_lsp then
+     configs.foo_lsp = {
+       default_config = {
+         cmd = {'/home/neovim/lua-language-server/run.sh'};
+         filetypes = {'lua'};
+         root_dir = function(fname)
+           return lspconfig.util.find_git_ancestor(fname)
+         end;
+         settings = {};
+       };
+     }
+   end
+
+- call `setup()` to enable the FileType autocmd
+>
+   lspconfig.foo_lsp.setup{}
+<
 
-* `:LspInfo` shows the status of active and configured language servers.
-The following support tab-completion for all arguments:
-* `:LspStart <config_name>` Start the requested server name. Will only successfully start if the command detects a root directory matching the current config. Pass `autostart = false` to your `.setup{}` call for a language server if you would like to launch clients solely with this command. Defaults to all servers matching current buffer filetype.
-* `:LspStop <client_id>` Defaults to stopping all buffer clients.
-* `:LspRestart <client_id>` Defaults to restarting all buffer clients.
 ==============================================================================
-10. The wiki                                              *lspconfig-the-wiki*
+COMMANDS                                                    *lspconfig-commands*
+
+- `:LspInfo` shows the status of active and configured language servers. Note
+  that client id refers to the neovim RPC instance connected to a given
+  language server.
 
+The following commands support tab-completion for all arguments. All commands
+that require a client id can either leverage tab-completion or the info
+contained in `:LspInfo`:
+
+- `:LspStart <config_name>` Launch requested configured client. Attempts to
+  detect a root directory for the given client and fails if not found.
+  Defaults to all *configured* servers matching the current filetype.
```

Comment:
Doesn't this actually mean "it will **only** work for a matching the current buffer `filetype` ". It's probably the same meaning, but I think it could be a bit more obvious that nothing will happen if there are no matching servers, even if you try to force it. e.g. `:LspStart clangd` in a javascript file.